common/workunit/workunit.cpp

```diff
@@ -1087,6 +1089,17 @@ class CDaliWorkUnit : public CPersistedWorkUnit
         VStringBuffer path("/GraphProgress/%s/%s", queryWuid(), graphName);
         return querySDS().connect(path, myProcessSession(), RTM_LOCK_WRITE|RTM_CREATE_QUERY, SDS_LOCK_TIMEOUT);
     }
+    IPropertyTree *getGraphProgressTree() const
+    {
+        IRemoteConnection *conn = queryProgressConnection();
```

Comment:
I think the same is true of all workunit data - it's not going to get updated unless the workunit is explicitly reloaded (and I also note that the workunit forceReload and reload methods don't appear to be implemented.

Adding a subscription mechanism would make this unimplementable by non-dali systems.

When thor reads subgraph status it does so for a bunch of graphs in sequence - it does not do so repeatedly for one graph to see if it changes (afaik). If it wanted to do the latter, we would need to add it to the subscribe mechanism implemented in one of my other recent PRs.